tests/unit/test_pipeline.py

```diff
@@ -139,29 +164,24 @@ def test_batch_validation():
 
 
 def check_expected_validation_exit(config, caplog, error_no, expected_msg):
-    try:
+    with pytest.raises(SystemExit) as e:
         Pipeline(config)
-    except SystemExit as e:
-        assert e.code == error_no
-        # We should only have one record
-        assert len(caplog.record_tuples) == 1
-        # Extract error message
-        msg = caplog.text.split("Validation errors found. Please see below.")[1].split(
-            "Validation errors found. Please see above."
-        )[0]
-        msg = re.sub("\n+", " ", msg)
-        msg = re.sub(" +", " ", msg).strip()
-        msg = re.sub("''", "'", msg)
-        # Check error types
-        expected_error_types = [error_type for error_type in expected_msg]
-        error_types = [x + " ERRORS" for x in msg.split(" ERRORS:")[0::2]]
-        assert set(error_types) == set(expected_error_types)
-        # Check actual messages. This is hacky; we remove expected substrings and
-        # ensure at the end that there is nothing subtantial left.
-        for error_type in error_types:
-            for schema in expected_msg[error_type]:
-                for str in expected_msg[error_type][schema]:
-                    msg = msg.replace(str, "")
-                msg = msg.replace(schema + ":", "")
-            msg = msg.replace(error_type + ":", "")
-        assert not any(ch.isalnum() for ch in msg)
+    assert e.value.code == error_no
+    # We should only have one record
+    assert len(caplog.record_tuples) == 1
+    # Extract error message
+    msg = caplog.text.split("Validation errors found. Please see below.")[1].split(
+        "Validation errors found. Please see above."
+    )[0]
+    msg = re.sub("\n+", " ", msg)
+    msg = re.sub(" +", " ", msg).strip()
+    msg = re.sub("''", "'", msg)
+    for error_type, schemas in expected_msg.items():
```

Comment:
I wonder if something like this could be extracted into the proposed `TestingUtils` project we were considering creating to house the FuzzyChecker.